pkg/occlient/occlient.go

```diff
@@ -1015,3 +973,189 @@ func (c *Client) ListRouteNames(labelSelector string) ([]string, error) {
+		Name:      volumeName,
+		MountPath: path,
+	},
+	)
+
+	log.Debugf("Updating DeploymentConfig: %v", dc)
+	_, err := c.appsClient.DeploymentConfigs(c.namespace).Update(dc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update DeploymentConfig: %v", dc)
+	}
+	return nil
+}
+
+// removeVolumeFromDC removes the volume from the given Deployment Config and
+// returns true. If the given volume is not found, it returns false.
+func removeVolumeFromDC(vol string, dc *appsv1.DeploymentConfig) bool {
+	found := false
+	for i, volume := range dc.Spec.Template.Spec.Volumes {
+		if volume.Name == vol {
+			found = true
+			dc.Spec.Template.Spec.Volumes = append(dc.Spec.Template.Spec.Volumes[:i], dc.Spec.Template.Spec.Volumes[i+1:]...)
+		}
+	}
+	return found
+}
+
+// removeVolumeMountFromDC removes the volumeMount from all the given containers
+// in the given Deployment Config and return true. If the given volumeMount is
+// not found, it returns false
+func removeVolumeMountFromDC(vm string, dc *appsv1.DeploymentConfig) bool {
+	found := false
+	for i, container := range dc.Spec.Template.Spec.Containers {
+		for j, volumeMount := range container.VolumeMounts {
+			if volumeMount.Name == vm {
+				found = true
+				dc.Spec.Template.Spec.Containers[i].VolumeMounts = append(dc.Spec.Template.Spec.Containers[i].VolumeMounts[:j], dc.Spec.Template.Spec.Containers[i].VolumeMounts[j+1:]...)
+			}
+		}
+	}
+	return found
+}
+
+// RemoveVolumeFromDeploymentConfig removes the volume associated with the
+// given PVC from the Deployment Config. Both, the volume entry and the
+// volume mount entry in the containers, are deleted.
+func (c *Client) RemoveVolumeFromDeploymentConfig(pvc string, dc *appsv1.DeploymentConfig) error {
+
+	volumeName, err := c.getVolumeFromPVC(pvc, dc)
+	if err != nil {
+		return errors.Wrap(err, "unable to get volume from PVC")
+	}
+
+	// Remove volume if volume exists in Deployment Config
+	if !removeVolumeFromDC(volumeName, dc) {
+		return fmt.Errorf("could not find volume '%v' in Deployment Config '%v'", volumeName, dc.Name)
+	}
+	log.Debugf("Found volume: %v in Deployment Config: %v", volumeName, dc.Name)
+
+	// Remove volume mount if volume mount exists
+	if !removeVolumeMountFromDC(volumeName, dc) {
+		return fmt.Errorf("could not find volumeMount: %v in Deployment Config: %v", volumeName, dc)
+	}
+
+	log.Debugf("Updating DeploymentConfig: %v", dc.Name)
+	_, err = c.appsClient.DeploymentConfigs(c.namespace).Update(dc)
+	if err != nil {
+		return errors.Wrapf(err, "unable to update the Deployment Config: %v", dc.Name)
+	}
+
+	return nil
+}
+
+// getVolumeFromPVC returns the name of the volume associated with the given
+// PVC in the given Deployment Config
+func (c *Client) getVolumeFromPVC(pvc string, dc *appsv1.DeploymentConfig) (string, error) {
+	for _, volume := range dc.Spec.Template.Spec.Volumes {
```

Comment:
There might be multiple volumes that are using the same PVC. Are we thinking about this?
There should be at least comment warning that this situation is theoretically possible.

Reply:
let's return all the volume names, and error out if more than one was returned

Reply:
made the changes